compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2793,6 +2795,36 @@ impl<'a> Parser<'a> {
         Ok((pat, expr))
     }
 
+    // Parses `cilk_for <src_pat> in <src_expr> <src_loop_block>` (`cilk_for` token already eaten).
+    fn parse_expr_cilk_for(&mut self, opt_label: Option<Label>, lo: Span) -> PResult<'a, P<Expr>> {
```

Comment:
Figure out if there's a way to deduplicate this since `parse_expr_for` is very similar. Then again, it's only 2 functions.